### DISCUSSION TITLE
请问如何筛选非美国 IP

### DISCUSSION THREAD

**Discussion by x151973** [XIU2/CloudflareSpeedTest#337] [讨论]
比如筛选次优亚洲线路
cf是莓果服务，cdn挨批全是美果的吗？
有些场景被访问端会凭壁掉米国挨批
速度能用就行，可用至上

**Comment by XIU2** [comment]
你这一堆谐音把我看傻了。。。

Cloudflare 有很多亚洲国家/地区的节点（根据官网所写：新加坡、日本、香港、韩国、泰国、印度等等），但总体数量较少，而且大都限制免费套餐的域名指向使用，剩下的一些可用的亚洲节点，Cloudflare 压根不分配给国内用户（移动倒是可以找到不少香港节点，但质量比较一般）。

你要知道 Cloudflare 的 IP 大部分都是 Anycast IP，也就是动态路由的，每个 IP 的路线和终点都不是固定的，不同地区的人访问同一个 IP，分配到的节点都是不一样的。

以新加坡节点为例，你在国内扫可能找不到几个新加坡的节点，但是如果你在新加坡的服务器上面扫，那就是另外一个结果了。因此不是你找不到 新加坡 节点，而是 Cloudflare 不给你用。

****

**电信联通**访问香港节点会**绕全球**，只有**移动直连**，移动用户测速什么都不用干，测速结果最前面的都是**延迟最低的香港节点**。

> 另外，虽然移动直连，但不代表一定速度快，我前面也说了**香港节点与国内内地之间线路**本身就**一般**，主要就是 Cloudflare 无法直接进入中国市场（只能靠和京东等公司间接合作），因此为了省钱就没有针对国内内地线路优化，所以默认分配的路由都是渣渣路线，所以才会促使我写出这个软件。。。

> 以前还有个大家能自选使用的**日本节点**（虽然是渣渣 NTT 线路，低峰期 70ms 高峰期爆炸），但是因为用的人太多了，今年初就被 Cloudflare 限制了，无法自选指向免费套餐的域名访问了。

****

另外，CloudflareST 在延迟测速后，
- 会先根据**丢包率排序**，
- 然后再根据**延迟排序**（不同丢包率的 IP 都分别独立按延迟排序），
- 最后再根据**下载速度排序**（如果没有 -dd 禁用下载测速的话），

因此最终结果最前面的就已经是**延迟最低、丢包最少**（或不丢包）的 IP 了，剩下的要么是**延迟高**、要么是**丢包高**，无论是哪点都对速度影响很大（特别是丢包）。

如果测试完成后，结果前列都是美国 IP，那么这已经是**最适合你、最快**的 IP 了。
